### PR TITLE
View user profiles

### LINKED ISSUE #72: On Logout change url to cuddly login url
Currently the URL remains the same after logging out, this means that no matter who logins in they will be taken to the last visited URL.

### PR DESCRIPTION
**Pull request by eeels22**
# Changes proposed

- Add `getUserById` and `listAllUsers` requests in backend
- Add `getUserById` and `getAllUsers` to `UserApi` in frontend
- Refactor `ProfilePage` to use a `Profile` component
- Amend path to logged in user's profile in `NavBar` to `/profile/mine`
- Add path param to `App.jsx` so we can access profiles by userId
- Amend root `/` page to display the Posts page
- Only show profile edit button if it belongs to the logged in user
- Resolves issue #72

## User story 62
`As a user of Cuddly, I want to view other users' profiles so I can learn more about them`

### Updated prompt links (bold on hover)
Screenshot 2021-05-15 at 13 27 13

### Profile (logged in user is Malin - path: `/profile/mine` - shows edit profile button)
Screenshot 2021-05-15 at 11 44 39
Screenshot 2021-05-15 at 11 44 27

### Profile (profile for Bob selected by userIid - path: `/profile/{id}` - does not show edit profile button)
Screenshot 2021-05-15 at 11 44 56
Screenshot 2021-05-15 at 11 45 04

### CODE REVIEW

File: frontend/src/components/navigation/Navbar.jsx [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-  const userURL = userObject;
-  const bellObject = require('../../assets/images/bell.svg');
-  const bellURL = bellObject;
-  const logoutObject = require('../../assets/images/logout.svg');
-  const logoutURL = logoutObject;
+  // Constants
+  const logoURL = require('../../assets/images/logo.svg');
+  const homeURL = require('../../assets/images/homepage.svg');
+  const userURL = require('../../assets/images/user.svg');
+  // const bellURL = require('../../assets/images/bell.svg');
```
Comment (NeuralAlchemist) [inline frontend/src/components/navigation/Navbar.jsx]: can you remove this line? possibly we dont have time for notifications

File: frontend/src/components/navigation/Navbar.jsx [hunk 2]
```diff
@@ -23,13 +21,13 @@ export default function Navbar({ onLogout }) {
...
-        <Link to="/profile">
+        <Link to="/profile/mine">
           <img className="user nav-icon" src={userURL} alt="Profile" />
         </Link>
 
-        <Link to="/notification">
+        {/* <Link to="/notification">
           <img className="notification" src={bellURL} alt="Notification" />
-        </Link>
+        </Link> */}
```
Comment (NeuralAlchemist) [inline frontend/src/components/navigation/Navbar.jsx]: can you remove this line? possibly we dont have time for notifications

File: frontend/src/components/profile/Profile.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  const hasLikedPosts = userLikesPostCards.length > 0;
+
+  return (
+    <div className="main-container-item ProfilePage">
+      <h1 className="page-name">Profile</h1>
+      <ProfileCard thisUser={thisUser} isLoggedInUser={isLoggedInUser}/>
+      <div className="user-feed">
+        <h3>Posts</h3>
+        {hasCreatedPosts && userPostCards}
+        {!hasCreatedPosts && <p className="prompt">You haven't created any posts yet</p>}
```
Comment (NeuralAlchemist) [inline frontend/src/components/profile/Profile.jsx]: Could we change it to say "Create posts to see them here" and maybe even add a link that is clickable and takes them to the post page?

File: frontend/src/components/profile/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <h3>Liked posts</h3>
+        {hasLikedPosts && userLikesPostCards}
+        {!hasLikedPosts && <p className="prompt">You haven't liked any posts yet</p>}
```
Comment (NeuralAlchemist) [inline frontend/src/components/profile/Profile.jsx]: Could we change it to say "Like posts to see them here" and maybe even add a link that is clickable and takes them to the post page?

**Review by deciocandido — COMMENTED** [review]
I could not see the display of all the users

File: frontend/src/components/profile/Profile.jsx (hunk 3 above)
Comment (eeels22) [reply frontend/src/components/profile/Profile.jsx]: great idea - updated

File: frontend/src/App.jsx [hunk 5]
```diff
@@ -24,32 +23,28 @@ export default function App() {
...
+      <BrowserRouter>
+        <Navbar onLogout={() => Auth.logout()} />
+
+        <div className="main-container">
+          <Switch>
+            <React.Suspense fallback={<div>Loading...</div>}>
+              <Route path="/posts">
+                <PostsPage />
+              </Route>
+              <Route path="/chat">
```
Comment (deciocandido) [inline frontend/src/App.jsx:35]: I suggest we remove the chat here, as it's not yet implemented yet

File: frontend/src/components/profile/Profile.jsx (hunk 4 above)
Comment (eeels22) [reply frontend/src/components/profile/Profile.jsx]: great idea - updated

**Comment by eeels22** [comment]
> I could not see the display of all the users

I was thinking that would be in scope of PBI 49. So the user never sees the full list of users, but when they search they get the users they are interested in. (I only implement the api method to get all users, not to display them)

So at the moment you can only view another user's profile by navigating to` profile/{userId}` but once we have the search implemented, that would display the users that match the criteria, but we still wouldn't need to display all users on the frontend

**Review by deciocandido — COMMENTED** [review]
Alright, it just that I saw on the UserApi.js have a getUserById and getAllUsers

**Comment by eeels22** [comment]
> Alright, it just that I saw on the UserApi.js have a getUserById and getAllUsers

Yea, I needed the list of all users to find the user specified by id to display their profile but that's the only use in this PR.

File: frontend/src/App.jsx (hunk 5 above)
Comment (eeels22) [reply frontend/src/App.jsx:35]: good idea

**Review by deciocandido — APPROVED** [review]
Great work

**Comment by NeuralAlchemist** [comment]
thanks! good work!

**Pull request merged**